exposure.py

```diff
@@ -0,0 +1,181 @@
+import math
+import csv
+import random
+from collections import OrderedDict
+from terminaltables import AsciiTable
+
+
+def parse_exposure_file(file_location):
+    """
+    :param file: File location
+    :return: Dictionary of exposures
+    { <name>: { min: <min>, max: <max> } }
+    """
+    exposures = []
+    with open(file_location, 'r') as filename:
+        reader = csv.DictReader(filename)
+        for row in reader:
+            if 'name' not in row or \
+               'min' not in row or \
+               'max' not in row:
+                raise Exception('''
+                    You must provide a min, max and name
+                    for each row - got {}.
+                    '''.format(row)
+                )
+            exposures.append({
+                'name': row['name'],
+                'min': float(row['min']),
+                'max': float(row['max']),
+            })
+
+    return exposures
+
+
+def get_exposure_args(existing_rosters, exposure_bounds, N, use_random,
+                      random_seed):
+    exposures = {}
+    for r in existing_rosters:
+        for p in r.players:
+            exposures[p.name] = exposures.get(p.name, 0) + 1
+
+    if use_random == 'y':
+        return get_exposure_args_random(exposures, existing_rosters,
+                                        exposure_bounds, N, random_seed)
+
+    return get_exposure_args_deterministic(exposures, existing_rosters,
+                                           exposure_bounds)
+
+
+def get_exposure_args_deterministic(exposures, existing_rosters,
+                                    exposure_bounds):
+    banned = []
+    locked = []
+
+    for bound in exposure_bounds:
+        name = bound['name']
+
+        total = float(len(existing_rosters) + 1)
+        min_lines = bound['min'] * total
+        max_lines = math.floor(bound['max'] * total)
+        lineups = exposures.get(name, 0)
+
+        if lineups < min_lines:
+            # TODO - downsize locked so solution is not impossible
+            locked.append(name)
+        elif lineups >= max_lines:
+            banned.append(name)
+
+    return {
+        'banned': banned,
+        'locked': locked,
+    }
+
+
+def get_exposure_args_random(exposures, existing_rosters, exposure_bounds, N,
```

Comment:
I think I lost this in a merge somewhere. It was counting the total exposure so that it's impossible to go over the max, but I think I like the simple (totally random) case better.